src/Microsoft.DotNet.CodeAnalysis/PackageFiles/codeAnalysis.targets

```diff
@@ -21,7 +21,18 @@
     <ValidPinvokeMappings Condition="'$(TargetGroup)'=='netcore50' or '$(TargetGroup)'=='netcore50aot' or '$(SupportsUWP)'=='true'">$(MSBuildThisFileDirectory)PinvokeAnalyzer_UWPApis.txt</ValidPinvokeMappings>
   </PropertyGroup>
 
-  <!-- If we have enabled analyzers we need to populate the right item groups -->
+  <PropertyGroup>
+    <!-- %24 = $ -->
+    <EnableFxCopAnalyzers Condition="$([System.Text.RegularExpressions.Regex]::IsMatch($(MSBuildProjectDirectory), 'src%24'))">true</EnableFxCopAnalyzers>
+    <EnableFxCopAnalyzers Condition="'$(EnableFxCopAnalyzers)' != 'true'">false</EnableFxCopAnalyzers>
+  </PropertyGroup>
+
+  <ItemGroup Condition="$(EnableFxCopAnalyzers)">
+    <Analyzer Include="$(CodeAnalyzerDir)*.dll" Exclude="$(CodeAnalyzerDir)Microsoft.DotNet.CodeAnalysis.dll"/>
+  </ItemGroup>
+  <PropertyGroup Condition="$(EnableFxCopAnalyzers)">
+    <CodeAnalysisRuleSet>$(MSBuildThisFileDirectory)Default.ruleset</CodeAnalysisRuleSet>
```

Comment:
If we add a condition on the CodeAnalysisRuleSet being already set you can have per-project overrides of the ruleset, which might come in handy.